os_fdp_adapters/xml-adapter/xml_to_fdp.py

```diff
@@ -0,0 +1,100 @@
+import sys
+import csv
+import json
+import tempfile
+import os
+import six
+
+sys.path.append('..')
+from common.fetchers import fetch_local_filename
+from common.urls import wrap
+
+csvDelimiter = ','
+
+
+def to_bytes(x):
+    if six.PY2 and type(x) is six.text_type:
+        return x.encode('utf8')
+    return x
+
+
+def gen_csv_file(url):
+    """
+    :param url: url can be the absolute address of an xml file, or  an xml file located at web (http://..../<name>.xml
+    :return: list of csv files, each contains a data-table extracted from the input xml file
+    """
+    orginBaseName = os.path.basename(url).strip(' ')
+    if url.find('http') == -1:
+        url = os.path.abspath(url)
+    with tempfile.NamedTemporaryFile(delete=False) as fd:
+        cmd = "xml2csv {} -a 135 >> {}".format(url, fd.name)
```

Comment:
Please use the `subprocess` python module and read the invocation's stdout instead of writing to a temporary file.